File: android/app/src/main/java/com/example/farmaciasdeguardiaensegovia/ui/components/NoPharmacyOnDutyCard.kt
/*
 * Copyright (C) 2025  Bruno Follon (@bFollon)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.example.farmaciasdeguardiaensegovia.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

/**
 * Card component for displaying "No pharmacy on duty" message
 * Equivalent to iOS NoScheduleView and ZBSScheduleView empty state
 */
@Composable
fun NoPharmacyOnDutyCard(
    modifier: Modifier = Modifier,
    message: String = "No hay farmacia de guardia programada para esta fecha.",
    additionalInfo: String? = "Intente refrescar o seleccione una fecha diferente."
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = Color(0xFFFDF8F3) // Very light orange/beige background matching iOS
        ),
        shape = RoundedCornerShape(8.dp)
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            // Warning icon and title
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                Icon(
                    imageVector = Icons.Default.Warning,
                    contentDescription = "Sin farmacia de guardia",
                    tint = Color(0xFFF2994A), // Orange color matching iOS
                    modifier = Modifier.size(20.dp)
                )
                Text(
                    text = "Sin farmacia de guardia",
                    style = MaterialTheme.typography.titleSmall,
                    fontWeight = FontWeight.Bold,
                    color = Color(0xFFF2994A) // Orange color matching iOS
                )
            }
            
            // Main message
            Text(
                text = message,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurface,
                textAlign = TextAlign.Start
            )
            
            // Additional info if provided
            additionalInfo?.let { info ->
                Text(
                    text = info,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.outline,
                    textAlign = TextAlign.Start
                )
            }
        }
    }
}

File: ios/FarmaciasDeGuardiaEnSegovia/Views/NoPharmacyOnDutyCard.swift
/*
 * Copyright (C) 2025  Bruno Follon (@bFollon)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import SwiftUI

/**
 * Card component for displaying "No pharmacy on duty" message inline
 * Equivalent to Android NoPharmacyOnDutyCard - shows within schedule content
 * rather than replacing the entire screen
 */
struct NoPharmacyOnDutyCard: View {
    let message: String
    let additionalInfo: String?

    init(
        message: String = "No hay farmacia de guardia programada para esta fecha.",
        additionalInfo: String? = "Intente refrescar o seleccione una fecha diferente."
    ) {
        self.message = message
        self.additionalInfo = additionalInfo
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            // Warning icon and title row
            HStack(spacing: 8) {
                Image(systemName: "exclamationmark.triangle.fill")
                    .foregroundColor(Color(hex: "F2994A"))
                    .frame(width: 20, height: 20)

                Text("Sin farmacia de guardia")
                    .font(.body)
                    .fontWeight(.bold)
                    .foregroundColor(Color(hex: "F2994A"))
            }

            // Main message
            Text(message)
                .font(.body)
                .foregroundColor(.primary)

            // Additional info if provided
            if let info = additionalInfo {
                Text(info)
                    .font(.subheadline)
                    .foregroundColor(.secondary)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(16)
        .background(
            RoundedRectangle(cornerRadius: 8)
                .fill(Color.orange.opacity(0.15))
        )
    }
}

// Helper extension for hex colors
extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3: // RGB (12-bit)
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6: // RGB (24-bit)
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8: // ARGB (32-bit)
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (255, 0, 0, 0)
        }
        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}

struct NoPharmacyOnDutyCard_Previews: PreviewProvider {
    static var previews: some View {
        VStack(spacing: 16) {
            NoPharmacyOnDutyCard()

            NoPharmacyOnDutyCard(
                message: "No hay farmacia de guardia asignada para Segovia Capital en esta fecha.",
                additionalInfo: "Por favor, consulte las farmacias de guardia de otras zonas cercanas o el calendario oficial."
            )
        }
        .padding()
    }
}
